kubernetes/schema_volume_source.go

```diff
@@ -218,6 +218,41 @@ func commonVolumeSources() map[string]*schema.Schema {
 				},
 			},
 		},
+		"csi": {
+			Type:        schema.TypeList,
+			Description: "Represents a CSI Volume. More info: http://kubernetes.io/docs/user-guide/volumes#csi",
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
```

Comment:
I see in the [API documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#csipersistentvolumesource-v1-core) for this resource there are several additional attributes for referencing secrets - is there a reason we haven't added them here?

Reply:
they are a bit tricky but i added them.
while the secret_ref state that `namespace` is optional. the api call fails if it not set. i extracted the `secret_ref` block to a common func (maybe better to extract to provider wide later)
i kept the argument optional as stated in the docs (to also make the block common). its a non breaking change as the namespace is also computed.